test/parallel/test-runner-wait-for.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+require('../common');
+const { test } = require('node:test');
+
+test('throws if condition is not a function', (t) => {
+  t.assert.throws(() => {
+    t.waitFor(5);
+  }, {
+    code: 'ERR_INVALID_ARG_TYPE',
+    message: /The "condition" argument must be of type function/,
+  });
+});
+
+test('throws if options is not an object', (t) => {
+  t.assert.throws(() => {
+    t.waitFor(() => {}, null);
+  }, {
+    code: 'ERR_INVALID_ARG_TYPE',
+    message: /The "options" argument must be of type object/,
+  });
+});
+
+test('throws if options.interval is not a number', (t) => {
+  t.assert.throws(() => {
+    t.waitFor(() => {}, { interval: 'foo' });
+  }, {
+    code: 'ERR_INVALID_ARG_TYPE',
+    message: /The "options\.interval" property must be of type number/,
+  });
+});
+
+test('throws if options.timeout is not a number', (t) => {
+  t.assert.throws(() => {
+    t.waitFor(() => {}, { timeout: 'foo' });
+  }, {
+    code: 'ERR_INVALID_ARG_TYPE',
+    message: /The "options\.timeout" property must be of type number/,
+  });
+});
```

Comment:
I think for readability reasons it would be nicer to have this inside a `describe`/`suite`, grouping the input validation tests

Reply:
That's fair. The CI is currently fully green though, so I would prefer not to go through that again to add one `suite()` call.

Reply:
I approved the PR, so this is non-blocking as far as I am concerned 🙂

Reply:
This is addressed since I pushed a change.